internal/pkg/template/diff/diff.go

```diff
@@ -90,7 +91,75 @@ func isYAMLLeaf(node *yaml.Node) bool {
 }
 
 func parseSequence(from, to *yaml.Node) (map[string]*Node, error) {
-	return nil, nil
+	fromSeq, toSeq := make([]yaml.Node, len(from.Content)), make([]yaml.Node, len(to.Content))
+	if err := to.Decode(&toSeq); err != nil {
+		return nil, err
+	}
+	if err := from.Decode(&fromSeq); err != nil {
+		return nil, err
+	}
+	cachedDiff, cachedErr := make(map[string]*Node), make(map[string]error)
+	lcsIdxInFrom, lcsIdxInTo := longestCommonSubsequence(fromSeq, toSeq, func(inA, inB int) bool {
+		diff, err := parse(&(fromSeq[inA]), &(toSeq[inB]), "")
+		if diff != nil { // NOTE: cache the diff only if a modification could have happened at this position.
+			cachedDiff[cachedKey(inA, inB)] = diff
+			cachedErr[cachedKey(inA, inB)] = err
+		}
+		return err == nil && diff == nil // NOTE: we swallow the error for now.
+	})
+	childKey := seqChildKeyFunc()
+	children := make(map[string]*Node)
+	var f, t, i int
+	for {
+		if i >= len(lcsIdxInFrom) {
+			break
+		}
+		lcsF, lcsT := lcsIdxInFrom[i], lcsIdxInTo[i]
+		switch {
+		case f == lcsF && t == lcsT: // Match.
+			f++
+			t++
+			i++
+			// TODO(lou1415926): (x unchanged items)
+		case f != lcsF && t != lcsT: // Modification.
+			// TODO(lou1415926): handle list of maps modification
+			diff, err := cachedDiff[cachedKey(f, t)], cachedErr[cachedKey(f, t)]
+			if err != nil {
+				return nil, err
+			}
+			children[childKey()] = diff
+			f++
+			t++
+		case f != lcsF: // Deletion.
+			children[childKey()] = &Node{oldValue: &(fromSeq[f])}
+			f++
+		case t != lcsT: // Insertion.
+			children[childKey()] = &Node{newValue: &(toSeq[t])}
+			t++
+		}
+	}
+	for {
```

Comment:
yeah some doc here that this for loop handles insertions/deletions/modifications _beyond_ the lcs would be helpful 👍

Reply:
Refactored the code a little, added the comment here! https://github.com/aws/copilot-cli/blob/eda0f814832fd689ab7e4448f09655a6eeddc3b8/internal/pkg/template/diff/iter.go#L50